app/upgrades/signetlaunch/data_signed_fps_test.go

```diff
@@ -0,0 +1,109 @@
+package signetlaunch_test
+
+import (
+	"bytes"
+	"encoding/json"
+	"testing"
+	"time"
+
+	"github.com/babylonlabs-io/babylon/app"
+	v1 "github.com/babylonlabs-io/babylon/app/upgrades/signetlaunch"
+	btcstktypes "github.com/babylonlabs-io/babylon/x/btcstaking/types"
+	tmproto "github.com/cometbft/cometbft/proto/tendermint/types"
+	"github.com/cosmos/cosmos-sdk/codec"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	"github.com/cosmos/cosmos-sdk/types/tx/signing"
+	"github.com/cosmos/cosmos-sdk/x/auth/ante"
+	authtx "github.com/cosmos/cosmos-sdk/x/auth/tx"
+	txmodule "github.com/cosmos/cosmos-sdk/x/auth/tx/config"
+	"github.com/stretchr/testify/require"
+)
+
+func TestValidateSignatureSignedFPsFromData(t *testing.T) {
+	bbnApp := app.NewTmpBabylonApp()
+	cdc := bbnApp.AppCodec()
+	// the chain ID in context needs to match the one used when creating the tx signature.
+	chainID := "bbn-1"
+
+	ctx := bbnApp.BaseApp.NewContextLegacy(true, tmproto.Header{Height: 1, ChainID: chainID, Time: time.Now().UTC()})
+	buff := bytes.NewBufferString(v1.SignedFPsStr)
+	simulateTx := false
+
+	var d v1.DataSignedFps
+	err := json.Unmarshal(buff.Bytes(), &d)
+	require.NoError(t, err)
+	require.Len(t, d.SignedTxsFP, 2)
```

Comment:
we probably do not need this assertion, as I imagine the length will be highly dependent on current launch context (devnet, testnet, mainnet)